### DISCUSSION TITLE
We made a awesome integrations with GitHub

### DISCUSSION THREAD

**Discussion by lawvs** [tolgee/tolgee-platform#1369] [Show and tell]
> Git integrations
> There are times when it would be great if localization data were synced with your Git repository. Or maybe you would like to create pull requests automatically when translations are changed...
> 
> To do this, integrations to GitHub, GitLab, or just support for Git itself will be helpful.

When we add new translate keys to the [AFFiNE](https://github.com/toeverything/AFFiNE) repo, GitHub Actions will check for key changes automatically.

<img width="1132" alt="Screen Shot 2022-09-12 at 11 54 01 PM" src="https://user-images.githubusercontent.com/18554747/189699969-5301b37d-a35f-41d8-8f34-788f34c4f633.png">

After merging the pull request, these keys will be added to the Tolgee platform (we can also add tags to unused keys).

This way, partners from the community can update the translations.

Finally, another GitHub Action will regularly synchronize the latest translation resources to the code repository (of course, you can also trigger it manually).

<img width="672" alt="Screen Shot 2022-09-13 at 12 02 18 AM" src="https://user-images.githubusercontent.com/18554747/189701669-a0c0d9ef-83b6-484b-be64-a931e5ac11f7.png">

You can find the detailed implementation here ->

https://github.com/toeverything/AFFiNE/tree/develop/libs/datasource/i18n

If anyone is interested in it, feel free to ask me for details.

**Comment by JanCizmar** [comment]
That's great folks! We want to support this natively hope it will be ready soon!

**Comment by tmlmt** [comment]
Any news on the native integration, @JanCizmar ?

I've found an up-to-date link to the relevant file of the Affine project where this was implemented (cc @lawvs): https://github.com/toeverything/AFFiNE/blob/master/packages/i18n/src/scripts/sync.ts

**Reply by JanCizmar** [reply]
Hi! Not yet. It's on the roadmap, hopefully we will get into that soon. https://github.com/orgs/tolgee/projects/2?pane=issue&itemId=20284734
